examples/fluids/problems/stats.c

```diff
@@ -0,0 +1,558 @@
+    PetscCall(DMProjectCoordinates(user->sp_stats.dm, NULL));  // Ensure that a coordinate FE exists
+  }
+
+  PetscCall(PetscObjectSetName((PetscObject)user->sp_stats.dm, "Spanwise_Stats"));
+  PetscCall(DMSetOptionsPrefix(user->sp_stats.dm, "spanstats_"));
+  PetscCall(DMSetFromOptions(user->sp_stats.dm));
+  PetscCall(DMViewFromOptions(user->sp_stats.dm, NULL, "-dm_view"));  // -spanstats_dm_view
+  {
+    PetscFE fe;
+    DMLabel label;
+
+    PetscCall(PetscFECreateLagrange(PETSC_COMM_SELF, problem->dim - 1, user->sp_stats.num_comp_stats, PETSC_FALSE, degree, PETSC_DECIDE, &fe));
+    PetscCall(PetscObjectSetName((PetscObject)fe, "stats"));
+    PetscCall(DMAddField(user->sp_stats.dm, NULL, (PetscObject)fe));
+    PetscCall(DMCreateDS(user->sp_stats.dm));
+    PetscCall(DMGetLabel(user->sp_stats.dm, "Face Sets", &label));
+
+    PetscCall(DMPlexSetClosurePermutationTensor(user->sp_stats.dm, PETSC_DETERMINE, NULL));
+    PetscCall(PetscFEDestroy(&fe));
+  }
+
+  PetscSection section;
+  PetscCall(DMGetLocalSection(user->sp_stats.dm, &section));
+  PetscCall(PetscSectionSetFieldName(section, 0, ""));
+  PetscCall(PetscSectionSetComponentName(section, 0, 0, "Mean Density"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 1, "Mean Pressure"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 2, "Mean Pressure Squared"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 3, "Mean Pressure Velocity X"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 4, "Mean Pressure Velocity Y"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 5, "Mean Pressure Velocity Z"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 6, "Mean Density Temperature"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 7, "Mean Density Temperature Flux X"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 8, "Mean Density Temperature Flux Y"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 9, "Mean Density Temperature Flux Z"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 10, "Mean Momentum X"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 11, "Mean Momentum Y"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 12, "Mean Momentum Z"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 13, "Mean Momentum Flux XX"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 14, "Mean Momentum Flux YY"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 15, "Mean Momentum Flux ZZ"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 16, "Mean Momentum Flux YZ"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 17, "Mean Momentum Flux XZ"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 18, "Mean Momentum Flux XY"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 19, "Mean Velocity X"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 20, "Mean Velocity Y"));
+  PetscCall(PetscSectionSetComponentName(section, 0, 21, "Mean Velocity Z"));
```

Comment:
We could have an enum with entries for every variable and use that in the qfunction, as in `v[TURB_MEAN_MOMENTUM_FLUX_YZ][i] = wdetJ * s.U.momentum[1] * s.Y.velocity[2];` so that we don't have so many magic numbers. The last placeholder in the enum can replace the magic number `22` that's currently in a couple places.

Reply:
I assume you've check that these visualize correctly in Paraview (at least the vectors; I think tensors aren't recognized by the PV/VTK reader for CGNS). Should we cut the spaces (with CamelCase) to match CGNS conventions?

Reply:
> We could have an enum with entries for every variable and use that in the qfunction, as in `v[TURB_MEAN_MOMENTUM_FLUX_YZ][i] = wdetJ * s.U.momentum[1] * s.Y.velocity[2];` so that we don't have so many magic numbers. The last placeholder in the enum can replace the magic number `22` that's currently in a couple places.

Yeah, that'd be a much nicer way to go about doing it. It think it'd also be nice for the statistics that are collected to be run-time selectable. That would require that the indexes be actual variables rather than enums. We can leave that for later, as it's not something we really need right now.

Reply:
> I assume you've check that these visualize correctly in Paraview (at least the vectors; I think tensors aren't recognized by the PV/VTK reader for CGNS). 

Yep, although it automatically interprets `Mean Momentum Flux X{X,Y,Z}` as a vector rather than leaving it alone.

> Should we cut the spaces (with CamelCase) to match CGNS conventions?

Ah, yep. Had it in my head that they had spaces rather than CamelCase.

Reply:
That's a bug in the CGNS reader that ships with VTK. There are a few improvements we should make sometime when our pants aren't on fire.

Reply:
I've implemented the enum now.

> The last placeholder in the enum can replace the magic number `22` that's currently in a couple places.

So the last placeholder would be `21`, not `22`. Should I create a new const variable of `TURB_MEAN_VELOCITY_Z + 1`, so just do that when setting `user->spanstats.num_comp_stats`?

Reply:
You can add `TURB_NUM_FIELDS` to the end of the enum and use that.